dotcom-rendering/src/web/components/Discussion.tsx

```diff
@@ -89,10 +89,23 @@ export const Discussion = ({
 
 	const handlePermalink = (commentId: number) => {
 		if (typeof window === 'undefined') return false;
-		window.location.hash = `#comment-${commentId}`;
+
+		const newHash = `#comment-${commentId}`;
+
+		// If the comment is already on the page, scroll to
+		// it directly. (If comment is not loaded, updating the hashCommentId
+		// should trigger setCommentPage, which will trigger a page load in
+		// discussion rendering.) We're adding scrolling here because React
+		// doesn't register a click as a change of state, if the same link is
+		// clicked twice. (Problem still persists for edge case of user navigating
+		// to a different page of comments without changing the comment id.)
+		document.querySelector(newHash)?.scrollIntoView();
```

Comment:
The existing scroll behaviour is controlled by the discussion-rendering app, and is triggered by the change of state triggered here (`setHashCommentId`). However, when a user clicks the same permalink twice, the second time doesn't register as a change of state and so doesn't trigger scrolling. In some ways it feels not-ideal for us to control scrolling from both this event handler and also from within the discussion App. But it seemed preferable for the time being (see PR description). Other views are very welcome, though!

https://github.com/guardian/discussion-rendering/blob/d034c4ab2e32d197cbd6da99c08cc4666550d4f1/src/App.tsx#L327-L334

Reply:
My confusion here is why do we have this coupling in the first place? Could this whole method be handled by `discussion-rendering`? It'd be good to know if this is the right place for this to live before we overload it.

This would have the knock on effect of having to move `getCommentContext` into `discussion-rendering`, but that also feels like a less leaky place for that to be.

If DCR were to decide if it is a permalink e.g. the hash is set to `#comment-{id}`, and just pass that `id` down to `discussion-rendering` and then it's all handled there, would that remove a bit of discussion specific code from DCR and make it easier to handle there?

Reply:
Yep, so if I remember correctly for longer-term solutions we discussed lifting the state up to discussion-rendering and exporting API utility functions for the parent to be able to construct the desired behaviour. The more I looked into it though, the more state there actually is in `discussion-rendering` itself, so I'm inclined to agree that moving `getCommentContext` down might be better than lifting the rest of the state up.

I guess one of the questions is whether we see `discussion-rendering` being used in any other contexts in the future, and if so what constraints that implies? e.g. allowing discussion to control scroll behaviour and also to set the window url might not be what we'd want in every context, so if it's going to be used outside DCR we'd need to think about how to make these behaviours optional, whilst also containing the logic for them within discussion-rendering?

Reply:
> whether we see `discussion-rendering` being used in any other contexts in the future

Good questions - @oliverlloyd @JamieB-gu @nicl - any information on this, I can't find any information [in the repo](https://github.com/guardian/discussion-rendering) or in the [architecture documents](https://github.com/guardian/dotcom-rendering/tree/main/dotcom-rendering/docs/architecture).

Reply:
My impression is that there's a lot of weight behind moving `getCommentContext` into discussion-rendering, this has come up a few times and would simplify things.

I think the original intention was to use discussion-rendering in apps-rendering which was what drove a lot of the dividing lines but I think it's fair to revisit this, maybe we can have `getCommentContext` in discussion-rendering with no issues even if this code were used elsewhere?

Reply:
> I think the original intention was to use discussion-rendering in apps-rendering
That's still the plan to the best of my knowledge.